scripts/fuse/README.MD

```diff
@@ -0,0 +1,166 @@
+
+The mocked attributes are:
+- the user and group (for the `user:` namespace)
+- working directory (for the `dir:` namespace)
+- environment variables (for the `proc:` namespace)
+- in the future: the process arguments
+
+
+### Example file system structure
+
+The filesystem structure mounted below the mountpoint could present as follows: (restructed to only one process with pid 41, with a maximum listing depth of 4)
+
+```
+|-- 41
+|   |-- cascading:
+|   |   |-- dir_and_file_at_once
+|   |   |   `-- leaf
+|   |   |-- dirkey
+|   |   |-- elektra
+|   |   |   |-- modules
+|   |   |   `-- version
+|   |   |-- info
+|   |   `-- person
+|   |       `-- name
+|   `-- user:
+|       |-- dir_and_file_at_once
+|       |   |-- @elektra.value
+|       |   `-- leaf
+|       |-- info
+|       `-- person
+|           `-- name
+
+```
+as obtained with
+```sh
+tree -L 4 mountpoint
+```
+
+
+
+## Assumptions and special behaviour
+
+Elektra does not map perfectly to the structure of a classical filesystem.
+Whenever deviations are necessary, the goal is to maximize usability and adhere to the expectations of common unix tools.
+
+### Intermediate directories
+
+In Elektra, there is no need to have a `path` of continous keys between ancestors.
+E.g. the existence of `user:/a` and `user:/a/b/c` does not imply the existence of `user:/a/b`.
+Such intermediate keys are mapped to "virtual" directories, s.t. navigation in an interactive manner is possible and the directory tree is continous.
+
+A side effect is that creating/deleting a single key can create/delete multiple "virtual" directories at once.
+Another is that copies created with `cp -r` will not have these vacuous intermediaries but instead empty keys.
+
+### Keys that act both as file and directory
+
+In Elektra, there are only keys and no distinction between directories and files are made. To enable a bidirectional mapping into a classical file system, the following mechanism is employed:
+
+If a key (`A`) exists, and has some child key (`B`), `A` is treated as a directory containing `B` and a special file `@elektra.value`. This file is used to access the value of `A`.
+
+Some side effects arise:
+E.g. recursively deleting directories with `rm -r` containing `@elektra.value` files raises (task uncritical) errors when attempting to delete a parent directory which was already deleted with the removal of an `@elektra.value` file directly below.
+
+### "Leaf"-Keys that act as as directories
+
+Keys at the bottom of the hierarchy are normally treaded as files. However in certain cases, like explicitly creating directories using `mkdir`, this would result in files beeing displayed, making it impossible to expand the hierarchy using standard tools.
+Therefore, when a directory is created, the special meta-key `fuse-directory` is set to override this behaviour.
+
+### Filesystem attributes
+
+The filesystem attributes are sourced from the file obtained by `kdb file`.
```

Comment:
Is it also possible to receive the file name itself? see #3599

What is the meaning of "number of links or directories", it seems to differ for the real file and the FUSE file?

Reply:
Since 66acb118f758035459212e3e730bc4e001d430b9 the functionality of `kdb file` is implemented natively.
Or do you propose the ability to query the file name for example as an xattr?

Reply:
Is this maybe also a candidate for a file below `®elektra`?

Reply:
See in https://github.com/ElektraInitiative/libelektra/pull/3657#discussion_r615417258.